libbeat/cmd/export/template.go

```diff
@@ -18,94 +18,50 @@
 func GenTemplateConfigCmd(settings instance.Settings) *cobra.Command {
 	genTemplateConfigCmd := &cobra.Command{
 		Use:   "template",
 		Short: "Export index template to stdout",
 		Run: func(cmd *cobra.Command, args []string) {
 			version, _ := cmd.Flags().GetString("es.version")
-			index, _ := cmd.Flags().GetString("index")
+			dir, _ := cmd.Flags().GetString("dir")
 			noILM, _ := cmd.Flags().GetBool("noilm")
 
-			b, err := instance.NewBeat(settings.Name, settings.IndexPrefix, settings.Version)
-			if err != nil {
-				fatalf("Error initializing beat: %+v", err)
-			}
-			err = b.InitWithSettings(settings)
-			if err != nil {
-				fatalf("Error initializing beat: %+v", err)
-			}
-
-			if version == "" {
-				version = b.Info.Version
-			}
-			esVersion, err := common.NewVersion(version)
-			if err != nil {
-				fatalf("Invalid Elasticsearch version: %+v", err)
+			if noILM {
+				settings.ILM = ilmNoopSupport
 			}
 
-			imFactory := settings.IndexManagement
-			if imFactory == nil {
-				imFactory = idxmgmt.MakeDefaultSupport(settings.ILM)
-			}
-			indexManager, err := imFactory(logp.NewLogger("index-management"), b.Info, b.RawConfig)
+			b, err := instance.NewInitializedBeat(settings)
 			if err != nil {
-				fatalf("Error initializing the index manager: %+v", err)
+				fatalfInitCmd(err)
 			}
 
-			tmplCfg, err := indexManager.TemplateConfig(!noILM)
-			if err != nil {
-				fatalf("Template error detected: %+v", err)
-			}
-			if tmplCfg.Enabled == false {
-				tmplCfg = template.DefaultConfig()
-			}
-
-			tmpl, err := template.New(b.Info.Version, index, *esVersion, tmplCfg, b.Config.Migration.Enabled())
-			if err != nil {
-				fatalf("Error generating template: %+v", err)
-			}
-
-			var templateString common.MapStr
-			if tmplCfg.Fields != "" {
-				fieldsPath := paths.Resolve(paths.Config, tmplCfg.Fields)
-				templateString, err = tmpl.LoadFile(fieldsPath)
-			} else {
-				templateString, err = tmpl.LoadBytes(b.Fields)
-			}
-			if err != nil {
-				fatalf("Error generating template: %+v", err)
-			}
-
-			_, err = os.Stdout.WriteString(templateString.StringToPrint() + "\n")
-			if err != nil {
-				fatalf("Error writing template: %+v", err)
+			clientHandler := idxmgmt.NewFileClientHandler(newIdxmgmtClient(dir, version))
+			idxManager := b.IdxSupporter.Manager(clientHandler, idxmgmt.BeatsAssets(b.Fields))
+			if err := idxManager.Setup(idxmgmt.LoadModeEnabled, idxmgmt.LoadModeDisabled); err != nil {
+				fatalf("Error exporting template: %+ver.", err)
 			}
 		},
 	}
 
 	genTemplateConfigCmd.Flags().String("es.version", settings.Version, "Elasticsearch version")
-	genTemplateConfigCmd.Flags().String("index", settings.IndexPrefix, "Base index name")
 	genTemplateConfigCmd.Flags().Bool("noilm", false, "Generate template with ILM disabled")
+	genTemplateConfigCmd.Flags().String("dir", "", "Specify directory for printing template files. By default templates are printed to stdout.")
```

Comment:
nice. Makes me wonder if it would make sense to have this as a positional parameter instead of a CLI flag. E.g. what happens when we introduce multiple templates support, where to write to if no dir is given?

Reply:
If no dir is given, we write to stdout.